### ISSUE TITLE
Read and persist `state` returned in `LocalDevice`/ `DeviceDetails`

### ISSUE THREAD

**Issue by ben-xD** [ably/ably-java#697]
In trying to make sure behaviour is correct in push notifications, I noticed that the `DeviceDetails.Push.state` is **always** null on Android and iOS. A user can get the LocalDevice by calling `AblyRealtime#device`, to get back LocalDevice, which extends from DeviceDetails, which contains `DeviceDetails.Push`. Unfortunately, `DeviceDetails.Push.state` is always nil.

It looks like it was never implemented? This value is not null when using the http api/ dashboard, its just the fact that ably-cocoa and ably-java doesn’t ever use it. This value is definitely returned to SDKs which call POST /push/deviceRegistrations. However, it is also returned by other HTTP requests, e.g. PUT rest.ably.io/push/deviceRegistrations/<deviceId>.

In ably-java, this is called in `ActivationStateMachine.java`, at `WaitingForPushDeviceDetails#transition`.

Similar ably-cocoa issue: https://github.com/ably/ably-cocoa/issues/1162

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1127) by [Unito](https://www.unito.io)

**Comment by ben-xD** [comment]
Please take a look at discussions at https://github.com/ably/ably-cocoa/issues/1162 before starting, it might need some API changes

**Comment by ben-xD** [comment]
Closing this for now, and a new issue will be created once the feature spec is updated https://github.com/ably/docs/issues/1251
